pkg/debug/start_debug.go

```diff
@@ -0,0 +1,163 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StartDebug(context *k8sutil.Context, clusterNamespace, deploymentName, flagValue string) {
+	deployment, err := verifyDeploymentExists(context, clusterNamespace, deploymentName)
+	if err != nil {
+		fmt.Printf("Missing mon or osd deployment name %s. %v\n", deploymentName, err)
+		os.Exit(1)
+	}
+
+	if flagValue != "" {
+		deployment.Spec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deployment.Spec.Template.Spec.Containers[0].LivenessProbe = nil
+	deployment.Spec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if flagValue != "" {
+		log.Printf("setting debug image to %s\n", flagValue)
+		deployment.Spec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deployment.Spec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deployment.Spec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(context, clusterNamespace, deployment); err != nil {
+		fmt.Printf("Failed to update deployment %s. %v\n", deployment.Name, err)
+		os.Exit(1)
+	}
+
+	deployment_Pod_Name, err := podRunningWithLabel(context, clusterNamespace, deployment.Spec)
+	if deployment_Pod_Name == "" || err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err := getAndUpdatetheScale(context, clusterNamespace, deployment.Name, 0); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deployment_Pod_Name)
+
+	err = waitForPodDeletion(context, clusterNamespace, deployment_Pod_Name)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	debug_deployment := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deployment.Spec,
+	}
+
+	debugDeployment, err := context.Clientset.AppsV1().Deployments(clusterNamespace).Create(ctx.TODO(), debug_deployment, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debug_deployment, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := getAndUpdatetheScale(context, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+}
+
+func getAndUpdatetheScale(context *k8sutil.Context, clusterNamespace, deploymentName string, scaleCount int) error {
```

Comment:
how about this name?
```suggestion
func setDeploymentScale(context *k8sutil.Context, clusterNamespace, deploymentName string, scaleCount int) error {
```

Reply:
changed